bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/item/ItemResource.java

```diff
@@ -367,6 +368,90 @@ public Response removeTag(@PathParam("itemname") String itemname, @PathParam("ta
         return Response.ok().build();
     }
 
+    /**
+     * Create or Update an item by supplying an item bean.
+     * @param itemname
+     * @param item the item bean.
+     * @return
+     */
+    @PUT
+    @Path("/{itemname: [a-zA-Z_0-9]*}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createOrUpdateItem(@PathParam("itemname") String itemname, ItemBean item) {
```

Comment:
Yes - i t works. My understanding is it resolves it based on the parameter types.
However, I'm happy to remove the other method(s). What do you want removed - just the other createOrUpdate method, or the others as well (ie we arguably don't need separate methods to set tags and groups as it can all be done with this method). I don't mind either way so I'll leave it to your guidance as maybe other UIs use these methods?